internal/agent/validate.go

```diff
@@ -31,16 +31,17 @@ func Validate(source string) error {
 		if err != nil {
 			return err
 		}
-		body, err := ioutil.ReadAll(resp.Body)
+		body, err := io.ReadAll(resp.Body)
 		if err != nil {
 			return err
 		}
 		//Convert the body to type string
 		yaml = string(body)
 	} else {
+		// Maybe we should just try to read the string for the normal headers? That would identify a full yaml vs a file
```

Comment:
This function is called `Validate` and most of the code is trying to figure out the type of source we are dealing with. The only validation we do is on line 63 below where we call `IsValid`.

Also, we rely on the error message to decide whether the string is a file or the source itself and I worry that different OSes may return different messages. Since we only run this on Linux, we are probably safe but we could probably use `os.Stat` to see if the file exists as a safer way.

Anyway, none of these problems was introduced in this PR so at least we don't make things worse than before.